docs/coding-guidelines.md

```diff
@@ -171,6 +171,29 @@ For example the following are correct:
     }
     ```
 
+1. Do not use `Func<>` or `Action<>` as return types. Instead, let the compiler convert the method into a func or action as necessary.
+
+    This is correct:
+    ```cs
+    Func<string, int> factory = DoSomething;
+
+    int DoSomething(string input)
+    {
+        // method body
+    }
+    ```
+
+    This is incorrect:
+    ```cs
+    Func<string, int> factory = GetFactory();
+
+    Func<string, int> GetFactory()
```

Comment:
What if I want a dynamically generated function? For example, a function to provide a logger in production and testing scenarios.

Reply:
The method that you want to change behaviour depending on testing or not testing accepts a `Func` or `Action` as an **input**. The production code in `src\` passes a method as this guideline suggests, and your test code in `test\` either passes in an anonymous method, or can also follow this guideline.

NuGet isn't a server app where prod/testing/dev environments traditionally exist, and uses Dependency Injection, not methods that return methods. I can't think of a good reason why a NuGet binary would want to change behaviour depending on "envionment".

We can discuss with more concrete examples if you like, but I don't fully understand the concern, or why a method returning a `Func` would be desirable, even in the example you gave.